File: backend/app/schemas/creative.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import datetime
from decimal import Decimal

# Base schemas
class CreativeProjectBase(BaseModel):
    title: str
    description: Optional[str] = None
    status: str = "active"

class DocumentAnalysisBase(BaseModel):
    analysis_type: str
    content: dict

class MarketAnalysisBase(BaseModel):
    analysis_data: dict

class IdeaGenerationBase(BaseModel):
    idea_content: dict
    evaluation_score: Optional[str] = None

class GeneratedImageBase(BaseModel):
    image_url: str
    prompt_used: str
    style_config: dict

class SolutionDesignBase(BaseModel):
    technical_solution: dict
    feasibility_study: dict
    implementation_path: dict

# Create schemas
class CreativeProjectCreate(CreativeProjectBase):
    user_id: str
    enterprise_id: str
    owner_id: int

class DocumentAnalysisCreate(DocumentAnalysisBase):
    project_id: int

class MarketAnalysisCreate(MarketAnalysisBase):
    project_id: int

class IdeaGenerationCreate(IdeaGenerationBase):
    project_id: int

class GeneratedImageCreate(GeneratedImageBase):
    project_id: int
    idea_id: int

class SolutionDesignCreate(SolutionDesignBase):
    project_id: int

class CreativeProjectUpdate(CreativeProjectBase):
    pass

# Response schemas
class CreativeProject(CreativeProjectBase):
    id: int
    owner_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        orm_mode = True

class DocumentAnalysis(DocumentAnalysisBase):
    id: int
    project_id: int
    created_at: datetime

    class Config:
        orm_mode = True

class MarketAnalysis(MarketAnalysisBase):
    id: int
    project_id: int
    created_at: datetime

    class Config:
        orm_mode = True

class IdeaGeneration(IdeaGenerationBase):
    id: int
    project_id: int
    created_at: datetime

    class Config:
        orm_mode = True

class GeneratedImage(GeneratedImageBase):
    id: int
    project_id: int
    idea_id: int
    created_at: datetime

    class Config:
        orm_mode = True

class SolutionDesign(SolutionDesignBase):
    id: int
    project_id: int
    created_at: datetime

    class Config:
        orm_mode = True 

File: frontend/src/types/creative.ts
export interface CreativeProject {
  id: string
  title: string
  description: string
  status: 'draft' | 'in_progress' | 'completed'
  created_at: string
  updated_at: string
  user_id: string
  enterprise_id: string
  tags: string[]
  members: string[]
  visibility: 'private' | 'team' | 'public'
}

export interface DocumentAnalysis {
  id: string
  project_id: string
  analysis_type: 'core_features' | 'feature_relations' | 'key_insights'
  content: Record<string, any>
  created_at: string
}

export interface MarketAnalysis {
  id: string
  project_id: string
  analysis_type: 'market_size' | 'competitor' | 'trend'
  analysis_data: Record<string, any>
  created_at: string
}

export interface IdeaGeneration {
  id: string
  project_id: string
  idea_content: Record<string, any>
  evaluation_score: number
  created_at: string
}

export interface GeneratedImage {
  id: string
  project_id: string
  idea_id: string
  image_url: string
  prompt_used: string
  style_config: Record<string, any>
  created_at: string
}

export type TimelineStatus = 'pending' | 'completed' | 'in_progress' | 'blocked'

export interface SolutionDesign {
  id: string
  project_id: string
  technical_solution: {
    architecture: Record<string, any>
    core_tech: Record<string, any>
    tech_stack: string[]
  }
  feasibility_study: {
    technical: Record<string, any>
    cost: Record<string, any>
    risks: Record<string, any>
  }
  implementation_path: {
    phases: Array<{
      name: string
      time: string
      status: TimelineStatus
      tasks: string[]
    }>
    resources: Record<string, any>
  }
  created_at: string
}

export interface ProjectMember {
  id: string
  name: string
  role: 'owner' | 'editor' | 'viewer'
} 